### PR TITLE
Allow the config path to be passed to uniffi-bindgen

### PR DESCRIPTION
**Pull request by eoger**
Fix https://github.com/mozilla/uniffi-rs/issues/304.

### CODE REVIEW

File: uniffi_bindgen/src/lib.rs [hunk 1]
```diff
@@ -252,27 +253,37 @@ pub fn run_tests<P: AsRef<Path>>(
...
-
-    let config_file = match config_file {
-        Ok(f) => f,
-        Err(_) => return Ok(default_config),
+    let config_file: Option<PathBuf> = match config_file_override {
+        Some(cfg) => Some(PathBuf::from(cfg)),
+        None => {
+            let path_guess = idl_file
+                .parent()
+                .ok_or_else(|| anyhow!("no parent!"))?
```
Comment (rfk) [inline uniffi_bindgen/src/lib.rs:268]: nit: please consider providing slightly more context in the error message here :-)

**Review by rfk — APPROVED** [review]
👍  thanks!

File: uniffi_bindgen/src/lib.rs [hunk 2]
```diff
@@ -252,27 +253,37 @@ pub fn run_tests<P: AsRef<Path>>(
...
+                .join("uniffi.toml");
+            match path_guess.canonicalize() {
+                Ok(f) => Some(f),
+                Err(_) => None,
+            }
```
Comment (jhugman) [inline uniffi_bindgen/src/lib.rs:273]: This assumes that the config file is only ever going to be `../uniffi.toml` relative to the `idl_file`.

File: uniffi_bindgen/src/lib.rs [hunk 3]
```diff
@@ -252,27 +253,37 @@ pub fn run_tests<P: AsRef<Path>>(
...
-        .expect("Missing $CARGO_MANIFEST_DIR, cannot load config file")
-        .into();
-
-    let config_file = pkg_path.join("uniffi.toml").canonicalize();
-
-    let config_file = match config_file {
-        Ok(f) => f,
-        Err(_) => return Ok(default_config),
+    let config_file: Option<PathBuf> = match config_file_override {
+        Some(cfg) => Some(PathBuf::from(cfg)),
```
Comment (jhugman) [inline uniffi_bindgen/src/lib.rs:264]: What happens when `config_file_override` is a relative path?

File: uniffi_bindgen/src/main.rs [hunk 4]
```diff
@@ -66,17 +78,25 @@ fn main() -> Result<()> {
             .arg(clap::Arg::with_name("cdylib_dir").required(true).help("Path to the directory containing the cdylib the scripts will be testing against."))
             .arg(clap::Arg::with_name("idl_file").required(true))
             .arg(clap::Arg::with_name("test_scripts").required(true).multiple(true).help("Foreign language(s) test scripts to run"))
+            .arg(
+                clap::Arg::with_name("config")
+                .long("--config-path")
+                .takes_value(true)
+                .help("Path to the optional uniffi config file. If not provided, uniffi-bindgen will try to guess from the IDL's file location.")
+            )
```
Comment (jhugman) [inline uniffi_bindgen/src/main.rs:86]: This arg is repeated for each subcommand. Nit: make the arg an arg of the `App`.

**Review by jhugman — COMMENTED** [review]
Drive by review of a quick fix. Thank you.

**Pull request merged**